### ISSUE TITLE
Scheduler/optimizer signature improvements

### ISSUE THREAD

**Issue by sarthakpati** [CBICA/GaNDLF#192]
**Is your feature request related to a problem? Please describe.**
Currently, any additional options getting passed on to initialize a scheduler need to be specified using specific variables, which makes the `get_scheduler` functions more complicated, and difficult to extend with new options.

**Describe the solution you'd like**
Put in a solution similar to preprocessing, augmentations and model code.
- Add new submodules: `scheduler` and `optimizer`
- Use the parameters from the config file. For example, take the following function in `GANDLF.parameterParsing`:

```python
def get_scheduler(
    which_scheduler, optimizer, batch_size, training_samples_size, learning_rate
):
    """
    This function parses the parameter from the config file and returns the appropriate object
    """
    step_size = 4 * batch_size * training_samples_size
    if which_scheduler == "triangle":
        clr = cyclical_lr(step_size, min_lr=10 ** -3, max_lr=1)
        scheduler_lr = LambdaLR(optimizer, [clr])
        print("Initial Learning Rate: ", learning_rate)
....
```
Can become:
```python
## file: GANDLF/scheduler/triangle.py

def base_triangle(parameters):
    """
    This function parses the parameters from the config file and returns the appropriate object
    """
    step_size = 4 * parameters["batch_size"] * parameters["training_samples_size"] # the latter needs to be added to the parameters in the training_loop, along with the optimizer
    # pick defaults for "min_lr", "max_lr" if not present in parameters
    # this should probably happen in parseConfig
    if not("min_lr" in parameters["scheduler"]):
        parameters["scheduler"]["min_lr"] = 10 ** -3
    if not("max_lr" in parameters["scheduler"]):
        parameters["scheduler"]["max_lr"] = 1
    clr = cyclical_lr(step_size, min_lr=parameters["scheduler"]["min_lr"], max_lr=parameters["scheduler"]["max_lr"])
    scheduler_lr = LambdaLR(optimizer, [clr])
    print("Initial Learning Rate: ", learning_rate)


## file: GANDLF/scheduler/__init__.py
from .triangle import base_triangle

global_schedulers_dict = {
    "triangle": base_triangle
}
```

And in `GANDLF.training_loop`, all scheulders can get initialized using a new `global_schedulers_dict`:
```python
scheduler = global_schedulers_dict[params["scheduler"]["name"]](params["scheduler"])
```

**Describe alternatives you've considered**
N.A.

**Additional context**
This will result in significant reduction in complexity of the code, and allow extensions to happen much quicker, which less code tie-ins between the defined configuration to the actual underlying functionality.
